xds/src/main/java/io/grpc/xds/client/ControlPlaneClient.java

```diff
@@ -399,15 +395,38 @@ private void handleRpcStreamClosed(Status error) {
       rpcRetryTimer = syncContext.schedule(
           new RpcRetryTask(), delayNanos, TimeUnit.NANOSECONDS, timeService);
 
-      checkArgument(!error.isOk(), "unexpected OK status");
-      String errorMsg = error.getDescription() != null
-          && error.getDescription().equals(CLOSED_BY_SERVER)
-              ? "ADS stream closed with status {0}: {1}. Cause: {2}"
-              : "ADS stream failed with status {0}: {1}. Cause: {2}";
-      logger.log(
-          XdsLogLevel.ERROR, errorMsg, error.getCode(), error.getDescription(), error.getCause());
+      Status newStatus = status;
+      if (responseReceived) {
+        // A closed ADS stream after a successful response is not considered an error. Servers may
+        // close streams for various reasons during normal operation, such as load balancing or
+        // underlying connection hitting its max connection age limit  (see gRFC A9).
+        if (!status.isOk()) {
+          newStatus = Status.UNAVAILABLE.withDescription(
```

Comment:
why isn't newStatus being set to `Status.OK`?  If you did that, then the if statement below and in XdsClientImpl could be simply (!newStatus.isOk()). Particularly for XdsClientImpl it would be much clearer.

Reply:
You could do `Status.OK.withDescription(`

Reply:
Updated. New status is `Status.OK` when a ADS stream is closed with an error but a response has been received.